### PR TITLE
Work around OpenCV highgui bug

### LINKED ISSUE #201: image_view: Attempt to unlock mutex that was not locked
Hi,

When I try to visualize an image using the image_view node and i get:

```
Attempt to unlock mutex that was not locked
Aborted
```

There were previous issues (#106) and (#153) but none seem to have a solution

Any clue?

My ROS version is Indigo on Arch Linux.

Thanks in advance.

### PR DESCRIPTION
**Pull request by roehling**
If OpenCV is compiled against GTK, running the GUI code in a dedicated thread will trigger a crash. While this is ultimately something for the OpenCV people to fix, we can work around the problem in the `image_view` node and fake a GUI thread with the help of  `ros::Timer`.

As a bonus, this PR also fixes the mutex assertion that occurs when the `image_view` program tries to shut down.

Fixes #201

### CODE REVIEW

**Comment by JWhitleyWork** [comment]
Tested as working and fixing the mutex issue.

**Comment by JWhitleyWork** [comment]
Can @vrabaud please review this? This is breaking image_view in Melodic.

**Comment by flixr** [comment]
You should use a `SteadyTimer` instead of a normal Timer (which could even pause on sim time).

**Comment by knorth55** [comment]
What is the status of this issue?
I'm suffering the same problem in #201 and fix it with this PR.

**Comment by JWhitleyWork** [comment]
The status is that the maintainers in ros-perception appear to have totally abandoned maintenance of ROS1 while working on ROS2. I have tried to bring some attention to the issues and PRs on this group of repositories (see https://groups.google.com/forum/m/#!topic/ros-sig-perception/pTA4tz2yEZU and https://discourse.ros.org/t/perception-sig-maintainers-still-maintaining/7065) but no one is answering. I'm not sure what else to do.

**Comment by knorth55** [comment]
@JWhitleyAStuff Thank you for the infomation. I also posted on ROSDiscourse for reviewing this PR.

**Comment by nitheeshkl** [comment]
@roehling Thanks for the fix! Meanwhile, since the nodelet also suffers from the same issue, it would be great if you include the same changes for nodelet as well. (instead of someone raising another issue & PR for the image nodelet in the future.)

**Comment by JWhitleyWork** [comment]
@nitheeshkl - Can you re-review now that @roehling has added the suggested changes?

**Review by nitheeshkl — APPROVED** [review]
Looks good to me. Verified this PR resolves the OpenCV crash issue at runtime.

**Comment by nitheeshkl** [comment]
@JWhitleyAStuff reviewed and verified on Melodic with Ubuntu 18.04 env. If only any of the maintainers can merge this PR now, would save a lot of people from wasting time on this bug!

**Comment by knorth55** [comment]
This PR is still waiting for reviewer response as described in
https://discourse.ros.org/t/perception-sig-maintainers-still-maintaining/7065.
I still want this PR to be merged because `image_view` is a fundamental feature.

Maintenance is a big problem, and  I also want to inform you ROS orphaned package project.
https://discourse.ros.org/t/ros-orphaned-package-maintainer-info-rep-152-for-review/7906

**Review by JWhitleyWork — APPROVED** [review]
LGTM and tested as working.

**Pull request merged**
